config/tctx.go

```diff
@@ -8,77 +8,69 @@ import (
 	"path/filepath"
 )
 
-type TLSConfig struct {
-	// Path to x509 certificate
-	CertPath string `json:"certPath"`
-	// Path to private key
-	KeyPath string `json:"keyPath"`
-	// Path to server CA certificate
-	CACertPath string `json:"caPath"`
-	// Disable tls host name verification (tls must be enabled)
-	DisableHostVerification bool `json:"disableHostVerification"`
-	// Override for target server name
-	ServerName string `json:"serverName"`
+type Tctx struct {
```

Comment:
Naming nit/idea... I think I prefer something like this or `ConfigReaderWriter` over `Tctx` in order to make it more clear that this can be used to read OR update tctx config.

```suggestion
type ConfigManager struct {
```